### ISSUE TITLE
computedAsync without `initialState` should return union T | `undefined`

### ISSUE THREAD

**Issue by aphex** [vueuse/vueuse#3926]
### Describe the bug

When using `computedAsync` in it's basic form:

```ts
const g = computedAsync(async () => {
  await new Promise((resolve) => setTimeout(resolve, 5000))

  return 'hello'
})
  ```

We get a type of `const g: Ref<string>`
It would be nice if we could get the type `const g: Ref<string | undefined>`

Without this the `template` assumes `g` will always be a `string` and TS has no issues.

For example something like this will throw a runtime error (`Unhandled error during execution of render function `) but not be caught by typescript
```html
<p>{{ g.charAt(1) }}</p>
```

### Reproduction

https://stackblitz.com/edit/vitejs-vite-hmsvky

### System Info

```Shell
System:
  OS: Windows 11 10.0.22631
  CPU: (32) x64 Intel(R) Core(TM) i9-14900K
  Memory: 32.46 GB / 63.75 GB
Binaries:
  Node: 18.20.0 - ~\AppData\Local\pnpm\node.EXE
  npm: 10.5.0 - ~\AppData\Local\pnpm\npm.CMD
  pnpm: 8.15.7 - ~\AppData\Local\pnpm\pnpm.EXE
Browsers:
  Edge: Chromium (123.0.2420.97)
  Internet Explorer: 11.0.22621.1
```

### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vueuse/vueuse/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vueuse/vueuse/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vueuse.org/guide).
- [X] Check that there isn't [already an issue](https://github.com/vueuse/vueuse/issues) that reports the same bug to avoid creating a duplicate.
- [X] Make sure this is a VueUse issue and not a framework-specific issue. For example, if it's a Vue SFC related bug, it should likely be reported to https://github.com/vuejs/core instead.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vueuse/vueuse/discussions).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by amw** [comment]
I can confirm this issue and I'd say it's a rather severe bug so I'm hoping it'll get some traction.

**Comment by ilyaliao** [comment]
Closed via #4207
